azure-pipelines/templates/linux-build.yml

```diff
@@ -23,8 +23,8 @@ steps:
     displayName: 'Build'
   ${{ if eq(parameters.Image, 'compile-farm-ci') }}:
     displayName: 'Build and test'
-  ${{ if eq(parameters.Tag, 'commit-checker') }}:
-    condition: and(succeeded(), eq(variables['Build.Reason'], 'PullRequest'))
+    # Run the docker task if it's not the commit-checker, or if it's a pull request
```

Comment:
Counter-suggestion: "Run the commit-checker only if it is a Pull Request"
These 'nots' and 'ors' confuse me :P